lib/poller.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const { EventEmitter } = require('events')
+
+class Poller extends EventEmitter {
+  constructor ({
+    pollingRateSeconds,
+    serializeSegments,
+    splitioApi
+  }) {
+    super()
+    this.cache = {}
+    this.pollingRateSeconds = pollingRateSeconds || 300
+    this.serializeSegments = serializeSegments || false
+    this.splitioApi = splitioApi
+    this.timeout = null
+  }
+
+  async pollForChanges () {
+    this.on('poll', async () => {
+      try {
+        const splitChanges = await this.splitioApi.getSplitChanges()
+        Object.assign(this.cache, splitChanges)
+      } catch (err) {
+        this.emit('error', err)
+      }
+      if (this.serializeSegments) {
```

Comment:
Should we move this `if` into the above `try` block? This would allow us to remove the duplicate
```
catch (err) {
  this.emit('error', err)
}
```